server/models/Schedule.js

```diff
@@ -0,0 +1,50 @@
+const mongoose = require("mongoose");
+
+const scheduleSchema = new mongoose.Schema({
+  profileId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "Profile",
+  },
+  name: {
+    type: String,
+    default: "",
+  },
+  days: [
```

Comment:
Let's say `min` and `max` indicate the days of the week, this design will allow the user to create redundant day schedules. Please see below for a suggestion:

```
const daySchedule = new mongoose.Schema({
  day: {
        type: Number,
        min: 1,
        max: 7,
        unique: true,
        required: true,
      },
  isAvailable: {
        type: Boolean,
        default: false,
      },
  start: {
        type: Number,
        min: 0,
        max: 23,
        required: true,
      },
  end: {
        type: Number,
        min: 0,
        max: 23,
        required: true,
      },
})

const scheduleSchema = new mongoose.Schema({
  profileId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: "Profile",
  },
  name: {
    type: String,
    default: "",
  },
  days: {
    Sunday: daySchedule,
    Monday: daySchedule,
    ........
}
})


```

Reply:
I just come across this problem and was thinking to limit the array size to 7. But your suggestion is great 😊.